### ISSUE TITLE
restarted container logs are not picked up in logs -f

### ISSUE THREAD

**Issue by cppforlife** [k14s/kapp#359]
**What steps did you take:**
i believe that when container is restarts (pod stays the same) logs are not retrieved as part of `kapp logs -f ...`. i think currently log tailing stops after container exits from the current run.

**Environment:**

- kapp version (use `kapp --version`): latest
- OS (e.g. from `/etc/os-release`): os x
- Kubernetes version (use `kubectl version`): 1.22

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by 100mik** [comment]
## To Reproduce
- Create the following manifest (`manifest.yaml`):
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: test
spec:
  containers:
  - name: command-demo-container
    image: debian
    command: ["bash","-c","for i in {1..5}; do echo -n 'Carvel\n'; sleep 1;  done"]
```
- Deploy the manifest
```bash
kapp deploy -a test -f manifest.yaml
```
- Tail logs for pods in the `test` app
``` bash
kapp logs -f -a test 
```
## Observed behaviour
`kapp` starts tailing the `pod`, waits for the pod to print "Carvel" 5 times before ending the tail. `kapp` does not exit.
```bash
Target cluster 'https://127.0.0.1:54856' (nodes: minikube)

# waiting for 'test > command-demo-container' logs to become available...
# starting tailing 'test > command-demo-container' logs
test > command-demo-container | Carvel
test > command-demo-container | Carvel
test > command-demo-container | Carvel
test > command-demo-container | Carvel
test > command-demo-container | Carvel
# ending tailing 'test > command-demo-container' logs
```
## What would be a better behaviour
- `kapp` notifies user about container exiting, reconciles to see if the container has restarted. `kapp` should tail the `Pod` if it is not deleted.
- `kapp` stops tailing and exits if there are no pods left to follow.

**Comment by cppforlife** [comment]
> kapp stops tailing and exits if there are no pods left to follow.

i dont think we would want this behaviour bit since pods may come up at a later point. (-f just keeps on watching forever. kapp logs without -f do end immediately.)

> kapp notifies user about container exiting, reconciles to see if pod is still present. kapp should tail the Pod if it is not deleted.

tiny nit, it's not that "pod is still present", it's more about container restarted.

**Comment by 100mik** [comment]
Definitely makes sense to keep watching for pods! Will make the change to the comment so that it is clear.

**Comment by renuy** [comment]
Available as part of kapp v0.47.0
